src/ledger/transaction.ts

```diff
@@ -79,14 +80,16 @@ export const addTransactionOnUsPayment = async ({
   payerUser,
   payeeUser,
   memoPayer,
-}) => {
+  shareMemoWithPayee,
+}: IAddTransactionOnUsPayment) => {
   const dealerPath = await dealerMediciPath()
 
   const entry = MainBook.entry(description)
 
   entry
     .credit(customerPath(payeeUser._id), sats * payeeUser.ratioBtc, {
       ...metadata,
+      memoPayer: shareMemoWithPayee ? memoPayer : null,
```

Comment:
ie: why is there a need to do that instead of just having a `memoPayer` that will just resolved as null if there is no memo?

Reply:
Because we want to have the sender's memo be shared with the receiver only if it's a pushpayment. If it's an invoice based payment, and we directly pass memoPayer, then the receiver's memo (if any) will get overwritten and not be returned as part of the `Wallet` query

Reply:
ok that makes sense. we'll need to document those edge cases when we'll be working on the API documentation